File: 2090-k-radius-subarray-averages/2090-k-radius-subarray-averages.py
class Solution:
    def getAverages(self, nums: List[int], k: int) -> List[int]:
        if len(nums)<2*k+1:
            return [-1 for i in range(len(nums))]
        S=sum(nums[:2*k+1])
        ans=[]
        ws=0
        we=2*k
        i=0
        while(i<len(nums)):
            if i<k:
                ans.append(-1)
                i=i+1
            elif i>=len(nums)-k:
                ans.append(-1)
                i+=1
            else:
                ans.append(int(S/(2*k+1)))
                if we+1<len(nums):
                    we+=1
                    S+=nums[we]
                S-=nums[ws]
                ws+=1
                i+=1
        return ans
                
            
        
                
            
                
            
        
            
        

File: 2090-k-radius-subarray-averages/2090-k-radius-subarray-averages.cpp
class Solution {
public:
    vector<int> getAverages(vector<int>& nums, int k) {
        if(k==0)return nums;
        int n=nums.size();
        long long currsum=0;
        vector<int>ans(n,-1);
        for(int i=0;i<=2*k and i<nums.size();i++)
            currsum+=nums[i];
        for(int i=0;i<n;i++)
        {
            if(i<k or n-k<=i)
            {
                ans[i]=-1;
            }
            else
            {
                ans[i]=currsum/(2*k+1);
                if(i+k+1<n)
                currsum+=nums[i+k+1];
                currsum-=nums[i-k];
            }
        }
        return ans;
    }
};